src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
-	dialers := b.dialers
+	var dialers []*dialer
+
+	// Checking destination port.
+	_, port, _ := net.SplitHostPort(addr)
+
+	// Are we attempting to connect to port 80 (plain HTTP)?
```

Comment:
Hey I think we're actually saying the same thing but just getting mixed up in the working --- I thought you were suggesting that limiting our servers to only proxy content to 80 and 8080 instead of to all the ports they currently proxy would limit this attack. In reality what will limit the attack is *not* sending port 80 content/unencrypted content to peers. I basically interpreted the following:

> only proxying ports such as 80 and 8080 on our servers since that would fix the potential Javascript script injection we're trying to close

as suggesting to only have our servers proxy those ports, but I believe you're suggesting to *send only those ports to our servers*, which is cool.